src/libos/src/vm/vm_area.rs

```diff
@@ -1,34 +1,88 @@
 use std::ops::{Deref, DerefMut};
 
+use super::page_tracker::PageTracker;
 use super::vm_perms::VMPerms;
 use super::vm_range::VMRange;
-use super::vm_util::FileBacked;
+use super::vm_util::{FileBacked, PagePolicy, VMMapOptions};
 use super::*;
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
 
+use super::vm_epc::EPC;
 use intrusive_collections::rbtree::{Link, RBTree};
 use intrusive_collections::{intrusive_adapter, KeyAdapter};
 
-#[derive(Clone, Debug, Default)]
+// Commit memory size when the PF occurs.
+// const COMMIT_ONCE_SIZE: usize = 256 * PAGE_SIZE;
+
+const COMMIT_ONCE_SIZE: usize = 4 * PAGE_SIZE;
+#[derive(Clone, Debug)]
 pub struct VMArea {
     range: VMRange,
     perms: VMPerms,
     file_backed: Option<FileBacked>,
+    initializer: Option<VMInitializer>, // Record the initializer for lazy intialization in PF handler. Drop when the VMA is fully committed
```

Comment:
I don't think VMInitializer needs to be saved in VMArea.

1. The memory can be initialized correctly according to the VMMapOption parameter when initializing the memory

2. When handling pagefault, newly allocated EPC memory is always zeroed memory (guaranteed by EAUG), so only the FileBacked case needs to be handled.

Reply:
Good suggestions. Done.